faunadb-scala/src/main/scala/faunadb/values/Value.scala

```diff
@@ -10,85 +10,117 @@ import java.time.{ LocalDate, ZonedDateTime, Instant }
 import scala.annotation.meta.{ param, field, getter }
 
 /**
-  * An abstract node in a FaunaDB response tree. Something conforming to this trait should be
-  * coerced into a concrete type before data can be accessed.
+  * A FaunaDB value.
+  *
+  * '''Reference''': [[https://faunadb.com/documentation/queries#values FaunaDB Values]]
+  *
+  * ===Overview===
+  *
+  * Value is an algebraic data type that represents the value of
+  * possible FaunaDB query responses. While it is possible to extract
+  * data out of a Value object using pattern matching, the
+  * [[faunadb.values.Field]] lets you create more complex and reusable
+  * data extractors.
   *
-  * @example Consider the Value modeling the root of the following tree:
   * {{{
-  * {
-  *   "ref": {"@ref": "some/ref" },
-  *   "data": { "someKey": "string1", "someKey2": 123 }
-  * }
+  * // Simple, adhoc extraction:
+  * value("data", "name").as[String].get
+  *
+  * // Using a reusable Field:
+  * val NameField = Field("data", "name").as[String]
+  * value(NameField).get
   * }}}
   *
-  * The data in this tree can be accessed using:
+  * Extraction can be composed:
+  *
   * {{{
-  *   value("ref").as[Ref].get // Ref("some/ref")
-  *   value("data", "someKey").as[String].get // "string1"
+  * val refAndNameAndAge = for {
+  *   ref <- value("ref").as[Ref]
+  *   name <- value("data", "name").as[String]
+  *   age <- value("data", "age").as[Int]
+  * } yield (name, age)
+  *
+  * refAndNameAndAge.get
+  *
+  * // or
+  *
+  * val RefAndNameAndAgeField = Field.zip(
```

Comment:
We are either missing the `ref` on the `yield` above or we need to remote the `ref` from this one. They sounds equivalent but the results are not the same.